### PR TITLE
Release v0.4.4

### PR DESCRIPTION
**Pull request by 12wrigja**
There is quite a lot to review here, and there's a couple more PRs that we should merge (and I'll update the release notes to include).

I wasn't quite sure where to put many of the changes in this release - observable changes in the order functions were called or properties were accessed ended up being something I skipped unless it sounded particularly important (e.g. sorted field access on property bags), but if those sorts of changes should be mentioned I'll have to take another pass over the commits to include the (many) relevant ones.

Similarly, there were lots of changes that are bug fixes, breaking changes, possibly both at the same time? Happy to reorganize the lists as people see fit - I don't have a good sense for whether fixing some of these bugs is likely to be a breaking change.

### CODE REVIEW

**Comment by 12wrigja** [comment]
Based on some offline feedback I've pruned/rewritten parts of the release notes to bundle together the various larger changes in this release.

I'd particularly appreciate feedback in deciding if some of the smaller items are worth mentioning. I've tried to remove things I think are really edge-case (observability / iteration ordering changes), but I've kept things I think might produce visible errors.

File: CHANGELOG.md
Comment (justingrant) [inline CHANGELOG.md:46]:
```suggestion
- Reject Z designators when parsing `Temporal.PlainTime` strings.
  This change avoids a common bug where a UTC timestamp is accidentally interpreted as if it were a local time in a real time zone.
  If you do want to parse the time portion of a UTC timestamp string, use: `Temporal.Instant.from(s).toZonedDateTimeISO('UTC').toPlainTime()`. ([a7a50eac])
```

File: CHANGELOG.md
Comment (justingrant) [inline CHANGELOG.md:47]:
```suggestion
- Reject 0-value components when parsing `Temporal.Duration` strings, and avoid rounding errors when nanoseconds components are present. ([58b5601a])
```

Comment (justingrant) [inline CHANGELOG.md:47]:
```suggestion
- Reject 0-value components when parsing `Temporal.Duration` strings, and avoid rounding errors when nanoseconds components are present. ([58b5601a])
```

File: CHANGELOG.md
Comment (justingrant) [inline CHANGELOG.md:48]:
```suggestion
- Reject `relativeTo` string options that are neither valid `Temporal.PlainDate` nor `Temporal.ZonedDateTime` strings, such as `'2022-08-18T17:01Z'`. ([4db15c41])
```

Comment (justingrant) [inline CHANGELOG.md:48]:
```suggestion
- Reject `relativeTo` string options that are neither valid `Temporal.PlainDate` nor `Temporal.ZonedDateTime` strings, such as `'2022-08-18T17:01Z'`. ([4db15c41])
```

File: CHANGELOG.md
Comment (justingrant) [inline CHANGELOG.md:50]:
```suggestion
- Validate required methods of `Temporal.TimeZoneProtocol` and `Temporal.CalendarProtocol`. ([84563cea], [755c7620], see also [Discussion](https://github.com/tc39/proposal-temporal/issues/2104#issuecomment-140))
```

**Review by justingrant — APPROVED** [review]
Wow, amazing work on this. I made a bunch of minor clarifying suggestions, and applied some general changes: 
* Put methods and class names in backticks
* Put `Temporal.` in front of all class names
* Replaced "timezone" with "time zone".

I made so many minor edits that I'm sure that some of them are wrong, so I'll probably want to take another review pass on this after these changes are merged. 

Thanks again for doing this!

File: CHANGELOG.md
Comment (12wrigja) [reply CHANGELOG.md:11]: This is making the commit ID both the text and the Markdown link specificer. When I render this markdown using the vscode markdown renderer they just appear as the commit ID, but linkified (to the commit on github).

Comment (12wrigja) [reply CHANGELOG.md:11]: (which works because the link specifier is then matched to the corresponding markdown link reference later at the bottom of the changelog).

**Comment by justingrant** [comment]
BTW, looks like the linter is complaining about something.

File: CHANGELOG.md
Comment (12wrigja) [reply CHANGELOG.md]: PTAL at the latest version and let me know what you think.

**Comment by 12wrigja** [comment]
Thanks so much for all the feedback!

I've incorporated all this feedback plus some small adjustments to guidance on how to handle the ZDT/DateTimeFormat interaction changes. PTAL.

File: CHANGELOG.md
Comment (justingrant) [inline CHANGELOG.md:13]:
```suggestion
         Be careful when caching the current system time zone (or an `Intl.DateTimeFormat` instance using that time zone), because the system time zone can change during the lifetime of your program (e.g. when a user on a mobile device crosses a time zone boundary or when a user manually edits time zone settings).
```

File: CHANGELOG.md
Comment (justingrant) [inline CHANGELOG.md:5]: What does "now works without `Intl.DateTimeFormat`" mean?

File: CHANGELOG.md
Comment (justingrant) [inline CHANGELOG.md:16]:
```suggestion
  - `Intl.DateTimeFormat` instances can be expensive to create. For performance-sensitive code where the same calendar and time zone are used repeatedly for formatting, we recommend creating and reusing an `Intl.DateTimeFormat` instance with the desired `timeZone` and `calendar` options, and then formatting using the value of `Temporal.ZonedDateTime.prototype.epochMilliseconds`.
```

File: CHANGELOG.md
Comment (justingrant) [inline CHANGELOG.md:18]:
```suggestion
- The `.calendar` property on Temporal types has been replaced with a `.calendarId` property that returns the string calendar identifier and a `.getCalendar()` method that returns an object that implements `Temporal.CalendarProtocol`.
```

File: CHANGELOG.md
Comment (justingrant) [inline CHANGELOG.md:20]:
```suggestion
  - Temporal will compare calendars by testing for string equality between calendar identifiers.
    When a calendar is an object, its `id` property will be used for comparison. ([3916e547], see also [Upstream issue](https://github.com/tc39/proposal-temporal/issues/1808))
```

File: CHANGELOG.md
Comment (justingrant) [inline CHANGELOG.md:22]:
```suggestion
- The `.timeZone` property on `Temporal.ZonedDateTime` has been replaced with a `.timeZoneId` property that returns the string time zone identifier and a `.getTimeZone()` method that returns an object that implements `Temporal.TimeZoneProtocol`. ([d3263f0], see also Upstream issue)
```

File: CHANGELOG.md
Comment (justingrant) [inline CHANGELOG.md:26]:
```suggestion
  - Temporal will compare time zones by testing for string equality between calendar identifiers.
    When a calendar is an object, its `id` property will be used for comparison.  ([d6cb2862], see also [Upstream issue](https://github.com/tc39/proposal-temporal/issues/1808))
```

File: CHANGELOG.md
Comment (justingrant) [inline CHANGELOG.md:40]:
```suggestion
- Fix an infinite-loop bug and a RangeError during non-ISO calendar calculations. ([d94c1cd9], see also [proposal-temporal PR](https://github.com/tc39/proposal-temporal/pull/2539), [Issue 1](https://github.com/tc39/proposal-temporal/issues/2383), [Issue 2](https://github.com/tc39/proposal-temporal/issues/2537))
```

File: CHANGELOG.md
Comment (justingrant) [inline CHANGELOG.md:57]: I would move this into the Breaking Changes section.

File: CHANGELOG.md
Comment (justingrant) [inline CHANGELOG.md:73]:
```suggestion
    - There can be multiple types of annotations in an IXDTF string.
      Temporal only recognizes time zone and calendar annotations.
      Unrecognized non-critical annotations will be ignored.
      Unrecognized critical annotations will cause the parsing method to throw an exception.
```

File: CHANGELOG.md
Comment (justingrant) [inline CHANGELOG.md:74]:
```suggestion
  - Allow `toString()` methods for `Temporal.PlainDate`, `Temporal.PlainDateTime`, `Temporal.PlainYearMonth`, `Temporal.PlainMonthDay`, and `Temporal.ZonedDateTime` to emit critical IXDTF annotations using the `'critical'` option.
```

File: CHANGELOG.md
Comment (justingrant) [inline CHANGELOG.md:76]:
```suggestion
    `calendarName: 'critical'` behaves like `calendarName: 'always'` and `timeZoneName: 'critical'` behaves like `timeZoneName: 'always'`, but they also output a `'!'` prefix in the corresponding annotation. ([50a64f16])
```

File: CHANGELOG.md
Comment (justingrant) [inline CHANGELOG.md:79]:
```suggestion
  - Allow calendar and/or time zone annotations after strings without a time part: YYYY-MM-DD, YYYY-MM, and MM-DD. ([acd6464f])
  - Disallow UTC offsets in YYYY-MM and MM-DD strings because they could cause ambiguity between an offset and the day of a YYYY-MM-DD string. ([acd6464f])
```
I think it's clearer if we split them.

File: CHANGELOG.md
Comment (justingrant) [inline CHANGELOG.md:85]: I would move this up to the top of the list, as I think we've gotten a lot of requests for this.

**Review by justingrant — APPROVED** [review]
Looks great!  Only a few minor suggestions this time. Thanks for putting in so much work to get this done... our users will be grateful too!

Do you want to go ahead and publish the release after this is merged?  Or are there other upstream PRs that we should wait to merge?

**Comment by 12wrigja** [comment]
I'm going to merge and create the release - I don't see any particularly interesting changes upstream that we need to pull in for this release in particular.

**Pull request merged**
